### PR TITLE
Fix GBDTs when `cat_features` is empty

### PR DESCRIPTION
**Pull request by xnuohz**

### CODE REVIEW

File: torch_frame/gbdt/tuned_lightgbm.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ class LightGBM(GBDT):
     def _to_lightgbm_input(
         self,
         tf: TensorFrame,
-    ) -> Tuple[np.ndarray, np.ndarray, List[int]]:
```
Comment (weihua916) [inline torch_frame/gbdt/tuned_lightgbm.py:21]: I think the return type of the last element should be pure `List[int]`

File: torch_frame/gbdt/tuned_catboost.py [hunk 2]
```diff
@@ -69,7 +69,8 @@ def _to_catboost_input(
             raise ValueError("The input TensorFrame object is empty.")
 
         df = pd.concat(dfs, axis=1)
-        cat_features = np.concatenate(cat_features, axis=0)
+        cat_features = np.concatenate(
+            cat_features, axis=0) if len(cat_features) else np.array([])
```
Comment (weihua916) [inline torch_frame/gbdt/tuned_catboost.py:73]: Can you help fix the return type to be `np.ndarray` (no optional)?

**Review by weihua916 — APPROVED** [review]
Thanks! LGTM after return type is fixed

File: torch_frame/gbdt/tuned_lightgbm.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ def _to_lightgbm_input(
                 concatenating tensors of numerical and categorical features of
                 the input :class:`TensorFrame`.
             y (numpy.ndarray): Prediction target :obj:`numpy.ndarray`.
-            cat_features (List[int]): Array containing indexes of
+            cat_features (List[int], optional): Array containing indexes of
```
Comment (weihua916) [inline torch_frame/gbdt/tuned_lightgbm.py:33]: same here. no optional

**Review by akihironitta — COMMENTED** [review]
LGTM! Could we also have a test case to avoid regressions in the future?

**Comment by xnuohz** [comment]
@weihua916 @akihironitta Fixed return types and improved test case.

**Review by weihua916 — APPROVED** [review]
LGTM!

**Pull request merged**